src/main/java/cph/sysint/libraryservice/control/TitleController.java

```diff
@@ -60,7 +49,17 @@ public ResponseEntity<GetTitleListResponse> getTitlesByPublisher(@PathVariable S
                                                                      @RequestParam(defaultValue = "5") int size) {
         Pageable pageable = PageRequest.of(page, size);
         GetTitleListResponse response = titleService.getByPublisher(publisher, pageable);
+        response.getTitles().forEach(titleDTO -> addLinksToTitle(titleDTO));
+        response.add(linkTo(TitleController.class).slash("publisher").slash(publisher).withSelfRel());
         return new ResponseEntity<>(response, HttpStatus.OK);
     }
 
+    private void addLinksToTitle(TitleDTO titleDTO) {
+        titleDTO.add(linkTo(TitleController.class).slash(titleDTO.getId()).withSelfRel());
+        titleDTO.getCategories().forEach(category -> {
+            titleDTO.add(linkTo(TitleController.class).slash(category).withRel("titles in '" + category + "' category"));
```

Comment:
is the `withRel` string going to be the actual key in the json?
If so, maybe they should not be with spaces since then it's harder to get them into things such as java objects of just indexing them in json

```javascript
obj.links["titles in 'horror' category"]
```

But maybe i'm misunderstanding what they mean?

Reply:
You are right. Here is new version, but I need feedback and ideas to how to represent those links 

![image](https://user-images.githubusercontent.com/54578672/198827458-b0347f1a-aca7-4bb6-872e-4088811216e1.png)

Reply:
Hmm i'm not sure tbh. I feel like if it's possible maybe all category recommendations should be an array and similarly with other recommendations.

So something like:
```json
{
  "_links": {
    "categoryRecommendations": [
      "firstlink",
      "secondlink"
    ]
  }
}
```

I'm not sure that's possible though?

Reply:
![image](https://user-images.githubusercontent.com/54578672/199114862-38c4986e-0665-4aa2-be6d-c17d15acff1b.png)
![image](https://user-images.githubusercontent.com/54578672/199115009-bb171ec3-24da-494f-84ca-00f0a33b8be2.png)